beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -206,6 +206,13 @@ private SafeFuture<Void> setExecutionData(
       final BeaconState blockSlotState,
       final BlockProductionPerformance blockProductionPerformance) {
 
+    if (spec.isMergeTransitionComplete(blockSlotState) && executionPayloadContext.isEmpty()) {
```

Comment:
I think we don't need the `isMergeTransitionComplete()` check. It's already covered by the `bodyBuilder.supportsExecutionPayload()` check a bit up the stack.

Reply:
bellatrix supports execution payload but it could be both pre merge and post merge